python/pyspark/sql/tests/test_dataframe_query_context.py

```diff
@@ -41,7 +41,7 @@ def test_dataframe_query_context(self):
                 error_class="DIVIDE_BY_ZERO",
                 message_parameters={"config": '"spark.sql.ansi.enabled"'},
                 query_context_type=QueryContextType.DataFrame,
-                pyspark_fragment="divide",
+                fragment="__truediv__",
```

Comment:
hmm, is it a regression?

Reply:
Ah, this is actually a bug fix. Previously it was (rouhgly) checking the Scala side `DataFrameQueryContext.fragment` which had to be `DataFrameQueryContext.pysparkFragement`. Now it correctly checks `DataFrameQueryContext.pysparkFragement`.

This is because we now merge `pysparkFragment` to `fragment` at `DataFrameQueryContext`, and it correctly tests the fragment.

Reply:
In the scala side, we use the call site that is closest to the user code as the fragment. I don't think `__truediv__` is user-friendly.

For Java, we can find the latest stack trace that is from `org.apache.spark`, whose next trace is the user code. Can we do the same thing in Python?

Reply:
Yeah, I think we should do. This PR doesn't change the original implementation though. I will create a new PR after this.

Reply:
Yeah, I think we should do. Let me take a look separately after this PR. This PR isn't related to the original implementation.

Reply:
Ah, actually it points out correctly. `/` is a shortcut of `__truediv__`, and it points out the line number properly. 
The missing part is to hide `pyspark.*`. I will take a separate look.

Reply:
cc @itholic

Reply:
I believe this is because the names are different in Python and Scala.

**Python**
https://github.com/apache/spark/blob/f0b7cfa56cb90ef70132d9656299956cbde00b53/python/pyspark/sql/column.py#L115

**Scala**

https://github.com/apache/spark/blob/f0b7cfa56cb90ef70132d9656299956cbde00b53/connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/Column.scala#L655


> Can we do the same thing in Python?

I think one way to do this is to add one `dict` into `with_origin` that can get the name of the Scala function corresponding to the Python function.

Reply:
Actually we can leverage `pyspark.util.try_simplify_traceback` but let's do this in a separate PR.